llvm/lib/Target/DirectX/DXILRootSignature.cpp

```diff
@@ -40,14 +40,64 @@ static bool reportError(LLVMContext *Ctx, Twine Message,
   return true;
 }
 
+static bool reportValueError(LLVMContext *Ctx, Twine ParamName, uint32_t Value,
+                             DiagnosticSeverity Severity = DS_Error) {
```

Comment:
Given that the function is called reportValue**Error** it seems weird for there to be a Severity parameter. We should either drop that parameter or rename this.